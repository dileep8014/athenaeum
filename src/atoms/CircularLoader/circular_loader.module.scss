@keyframes loader {
  0% {
    transform: rotate(0deg);
  }

  50% {
    transform: rotate(180deg);
  }

  100% {
    transform: rotate(360deg);
  }
}

.circular-loader {
  animation-duration: .6s;
  animation-iteration-count: infinite;
  animation-name: loader;
  animation-timing-function: linear;
  border-bottom: 3px solid color('neutral-1');
  border-left: 3px solid transparent;
  border-radius: 100%;
  border-right: 3px solid color('neutral-1');
  border-top: 3px solid transparent;
  height: rem-calc(24px);
  width: rem-calc(24px);
  left: calc(50% - 12px);
  position: relative;
  transform: translateY(-50%);
}
