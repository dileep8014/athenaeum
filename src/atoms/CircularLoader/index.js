import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './circular_loader.module.scss';

function CircularLoader(props) {
  const {
    className,
    ...rest
  } = props;

  const classes = classnames(
    styles['circular-loader'],
    className,
  );

  return (
    <div className={classes} {...rest} />
  );
}

CircularLoader.propTypes = {

  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
};

export default CircularLoader;
