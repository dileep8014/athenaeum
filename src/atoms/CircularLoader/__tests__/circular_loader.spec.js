import React from 'react';
import renderer from 'react-test-renderer';

import CircularLoader from 'atoms/CircularLoader';

describe('<CircularLoader />', () => {
  it('renders correctly', () => {
    const actual = renderer.create(<CircularLoader />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
