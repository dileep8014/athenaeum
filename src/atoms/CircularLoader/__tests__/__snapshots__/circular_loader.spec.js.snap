// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircularLoader /> renders correctly 1`] = `
<div
  className="circular-loader"
/>
`;
